src/components/common/Extensions/NotificationBanner/types.ts

```diff
@@ -1,20 +1,29 @@
 import { MessageDescriptor } from 'react-intl';
+import { STATUS_TYPES } from '~v5/shared/StatusText/types';
 
 export interface NotificationBannerProps {
-  status: StatusType;
+  status: (typeof STATUS_TYPES)[keyof typeof STATUS_TYPES];
   title?: string | MessageDescriptor;
-  actionText?: string | MessageDescriptor;
-  actionType?: ActionType;
-  redirectUrl?: string;
   isFullSize?: boolean;
   isAlt?: boolean;
-  onClick?: () => void;
+  action?:
+    | {
+        type: 'copy';
+        copyContent: string;
+        actionText: string | MessageDescriptor;
+      }
+    | {
+        type: 'redirect';
+        href: string;
+        actionText: string | MessageDescriptor;
+      }
+    | {
+        type: 'call-to-action';
+        onClick: () => void;
+        actionText: string | MessageDescriptor;
```

Comment:
If you want to have `actionText` in all types just add `{ actionText: React.ReactNode } & ( ...union of types... )`  before the union

Reply:
Changed